Add ux documentation - part 1 (#185)
SchwarzIT/onyx: apps/docs/src/basics/typography.md

# Typography

::: danger
Still content to create!
:::

> The onyx font system includes three different font families that can be used throughout your application, pending on the usecase. It contains several headline styles as well as paragraph styles and mono space styles.

Reviewer: ```suggestion
The onyx font system includes two different font families that can be used throughout your application, pending on the usecase. It contains several headline styles as well as paragraph styles and mono space styles.
```

Author: Why two?
Source Sans, Lidl and Kaufland

Author: okay, saw it:)

::: info
The font families that can be used with onyx are:

- [Source Sans 3](https://fonts.google.com/specimen/Source+Sans+3)
- [Source Code Pro](https://fonts.google.com/specimen/Source+Code+Pro?query=source)
- [Lidl Font Pro](https://www.brand.lidl/brandelements/brandelements/view/id/664#headline_2486)

> - [Kaufland Regular](https://google.com)

Reviewer: Until we have a decision to make lidl/kaufland etc. theme public I'd like to remove it here

Reviewer: Also remove all content below that references lidl or kaufland

Author: I would write "lidl and kaufland font to come later". I would not delete it as a whole

  :::

## Brand Typography

> The standard font family is defined to be the Source Sans. If you are creating a new onyx project, this will always be the startingpoint. In the initial phase the Source Sans applies to all color themes that are provided by onyx.

Reviewer: ```suggestion
The default font family is defined to be the Source Sans and Source Code Pro (monospace). If you are creating a new onyx project, this will always be included out-of-the-box. In the initial phase, Source Sans applies to all color themes that are provided by onyx.
```

In case you are working on a Kaufland application or Lidl application, you are free to switch into the brand font families of Kaufland Regular or Lidl Font Pro.

::: warning Attention
Please note that only the `Kaufland Regular` may be implemented in the Kaufland theme and only the `Lidl Font Pro` in the Lidl theme!

For all the other color themes, `Source Sans` remains the font of choice!
:::

### How to import

To import Kaufland Regular and Lidl Font Pro follow the steps below.

::: code-group

```txt [Kaufland Regular]
Lorem Ipsum
```

```txt [Lidl Font Pro]
Lorem Ipsum
```

:::

## Font stacks

_custom component for displaying font definitions_

> ### Headlines

Reviewer: Can we replace all those details / accordions when we have the custom typography component?

Author: Yes this is the intention.

::: details h1

This is the page title. It can only be used once per page at the very top.

#### Details

| Property            |          Value |
| ------------------- | -------------: |
| `weight`            | SemiBold - 600 |

> | `size`              |          28 px |

Author: fine for me

## Writing style

If you want to use text in your application, please take the following principles into account.

::: details Focus

Formulate your text purposefully and concisely. Do not use unnecessary verbiage. Nevertheless, form short, complete and grammatically correct sentences.

:::

::: details Descriptive

Describe prompts clearly and use the imperative (especially with buttons). Buttons are always titled with verbs and should at best consist of only one word. More than one word is possible in exceptional cases.

:::

::: details Hierarchy

Hierarchy is key to a well working user experience. Always be aware on how to use section titles and element headlines.
Also use [color and contrast](/basics/colors) for creating hierarchy in a subtle way.

:::

::: details Context

Texts must always have a connection to the things that are going on in the screen, in terms of content.
Never text about content, the user cannot relate to.

:::

::: details Consistency

> Texts never are built with uppercases. Always use the provided line hight for spacing. Do not try to squeeze characters or textblocks. Less is more - Minimalism is key.

Reviewer: ```suggestion
Texts should never be built in uppercase. Always use the provided line height for spacing. Do not try to squeeze characters or text blocks. Less is more - Minimalism is key.
```

:::

## Links

Links are the tool of choice to refer to another or external page. Please use the `link` font style. By default links have a underline and are primary-colored. Under consideration of the text styling principles the text-color can be changed in exceptional usecases. Please doublecheck with your assigned UX-Designer.

## Truncation

> For more details about truncation and how to use it, please visit the [accessibility foundations](/basics/accessibility).

Reviewer: ```suggestion
For more details about truncation and how to use it, please visit the [truncation basics](/basics/truncation).
```
